tests/e2e/specs/app-ratings-overview/experience-rating-banner.test.js

```diff
@@ -0,0 +1,298 @@
+import AppRatingsOverview from '../../utils/app-ratings-overview';
+
+test.use( { storageState: process.env.ADMINSTATE } );
+
+test.describe.configure( { mode: 'serial' } );
+
+/**
+ * @type {import('../../utils/app-ratings-overview').default} appRatingsOverview
+ */
+let appRatingsOverview = null;
+
+/**
+ * @type {import('@playwright/test').Page} page
+ */
+let page = null;
+const BANNER_CLASS = '.gla-experience-rating-banner';
+
+test.describe( 'App Ratings Banner', () => {
+	test.beforeAll( async ( { browser } ) => {
+		page = await browser.newPage();
+		appRatingsOverview = new AppRatingsOverview( page );
+		await Promise.all( [
+			appRatingsOverview.mockRequests(),
+			setOnboardedMerchant(),
+		] );
+		await appRatingsOverview.goto();
+	} );
+
+	test.afterAll( async () => {
+		await clearOnboardedMerchant();
+		await page.close();
+	} );
+
+	test.describe( 'Banner is visible across all tabs', () => {
+		test.beforeAll( async () => {
+			await page.waitForSelector( '.gla-dashboard', {
+				state: 'visible',
+			} );
+		} );
+
+		test.afterAll( async () => {
+			await page.evaluate( 'window.localStorage.clear()' );
+		} );
+
+		test( 'Banner visibility on dashboard tab', async () => {
+			await page.waitForSelector( '.gla-dashboard', {
+				state: 'visible',
+			} );
+
+			const banner = page.locator( BANNER_CLASS );
+			await expect( banner ).toBeVisible();
+		} );
+
+		test( 'Banner visibility on product feed tab', async () => {
+			await appRatingsOverview.goto(
+				'/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed'
+			);
+			await page.waitForSelector( '.gla-product-feed', {
+				state: 'visible',
+			} );
+
+			const banner = page.locator( BANNER_CLASS );
+			await expect( banner ).toBeVisible();
+		} );
+
+		test( 'Banner has action buttons', async () => {
+			const banner = page.locator( BANNER_CLASS );
+			const goodButton = banner.getByRole( 'button', {
+				name: 'Good',
+			} );
+			const needHelpButton = banner.getByRole( 'link', {
+				name: 'Need help',
+			} );
+
+			await expect( goodButton ).toBeVisible();
+			await expect( needHelpButton ).toBeVisible();
+		} );
+
+		test( 'Need help button opens external link', async () => {
+			const banner = page.locator( BANNER_CLASS );
+			const needHelpButton = banner.getByRole( 'link', {
+				name: 'Need help',
+			} );
+
+			await expect( needHelpButton ).toHaveAttribute(
+				'href',
+				'https://woocommerce.com'
+			);
+			await expect( needHelpButton ).toHaveAttribute(
+				'target',
+				'_blank'
+			);
+		} );
+
+		test( 'Hides the banner when dismissed', async () => {
+			await appRatingsOverview.fulfillUsersPreferences();
+
+			const banner = page.locator( BANNER_CLASS );
+			const dismissButton = banner.getByRole( 'button', {
+				name: 'Close',
+			} );
+
+			await dismissButton.click();
+
+			await expect( banner ).not.toBeVisible();
+		} );
+
+		test( 'Banner is not visible after reload once dismissed', async () => {
+			await page.reload();
```

Comment:
We probably need to wait for something else on the page to be visible before checking that the banner is not visible or else this could result in a false positive.

Reply:
Resolved in 6b879a0e234